### ISSUE TITLE
Database query filter types not corresponding to the API specs

### ISSUE THREAD

**Issue by JavierCane** [makenotion/notion-sdk-js#77]
Hi!

Firstly, congrats for the nice work on the DSK types DSL modelling allowing exhaustive "pattern matching" 🙌

Said that, I am not sure if this is a bug or I am missing something, let me explain it with an example 😊

# First try building a query: Types based

```ts
const queryFilterSelectFilterTypeBased: SelectFilter = {
    property: "Status",
    equals: "Blocked"
}

this.client.databases.query({
    database_id: this.kanbanDatabaseId,
    filter: queryFilterSelectFilterTypeBased
});
```

Log:

```bash
building the error
Client warn: request fail {
  code: 'validation_error',
  message: 'body failed validation. Fix one: body.filter.or should be defined, instead was `undefined`. body.filter.and should be defined, instead was `undefined`. body.filter.title should be defined, instead was `undefined`. body.filter.text should be defined, instead was `undefined`. body.filter.number should be defined, instead was `undefined`. body.filter.checkbox should be defined, instead was `undefined`. body.filter.select should be defined, instead was `undefined`. body.filter.multi_select should be defined, instead was `undefined`. body.filter.date should be defined, instead was `undefined`. body.filter.people should be defined, instead was `undefined`. body.filter.files should be defined, instead was `undefined`. body.filter.url should be defined, instead was `undefined`. body.filter.email should be defined, instead was `undefined`. body.filter.phone should be defined, instead was `undefined`. body.filter.relation should be defined, instead was `undefined`. body.filter.created_by should be defined, instead was `undefined`. body.filter.created_time should be defined, instead was `undefined`. body.filter.last_edited_by should be defined, instead was `undefined`. body.filter.last_edited_time should be defined, instead was `undefined`. body.filter.formula should be defined, instead was `undefined`. body.filter.rollup should be defined, instead was `undefined`.'
}
```

Unexpected result:

- `building the error` log output: Already being handled in https://github.com/makenotion/notion-sdk-js/issues/48
- Validation error: It seems to be expecting a key inside the `body.filter` defining the property type

## A little SDK research

Taking a look to the `SelectFilter` type, it is being defined as:

```ts
export interface SelectFilter extends SinglePropertyFilterBase {
    equals?: string;
    does_not_equal?: string;
    is_empty?: true;
    is_not_empty?: true;
}

export interface SinglePropertyFilterBase {
    property: string;
}

```

So, at the end, it will result in constructing an object like the previous one:

```json
{
    "property": "Status",
    "equals": "Blocked"
}
```

And looking at how the Notion SDK client implements the `databases.query` operation, it would be passing through this very same object to the HTTP endpoint without any previous modification:

https://github.com/makenotion/notion-sdk-js/blob/main/src/Client.ts#L183

# Second try: HTTP Spec based

Taking a look to [the Filter object documentation in the API docs](https://developers.notion.com/reference/post-database-query#post-database-query-filter), it seems to be having an additional nesting level specifying the property type of the property being queried. `text` in the documented example:

```json
{
  "property": "Landmark",
  "text": {
    "contains": "Bridge"
  }
}
```

It seems that the SDK would not be adding that additional level, so lets try executing the following, even if TypeScript produces a compiling error:

```ts
const statusFilterHttpSpecsBased: Filter = {
    property: "Status",
    select: { equals: status }
};

this.client.databases.query({
    database_id: this.kanbanDatabaseId,
    filter: statusFilterHttpSpecsBased
});
```

It works! 🎉

Despite the obvious mismatch at types level for TypeScript:

```bash
TS2322: Type '{ property: string; select: { equals: string; }; }' is not assignable to type 'Filter'.   Object literal may only specify known properties, and 'select' does not exist in type 'Filter'.
```

# Conclusion

I could be missing anything, so please, feel free to point me what is it if I am mistaken 🙏

If not, which approach would you prefer in order to fix it?

Thanks!

**Comment by Zchary-Ma** [comment]
Struggling against the same problem, is there any clue?

**Comment by blackmad** [comment]
This sounds like the typescript typings might be slightly off, let me look this AM

**Comment by blackmad** [comment]
This should be fixed in 0.1.7 which I just released - https://github.com/makenotion/notion-sdk-js/pull/98
